### PR TITLE
Removed Work.merge_into

### LINKED ISSUE #241: Get rid of Work.was_merged_into_id
Our work merging code currently keeps the old work around, even though we no longer need it. We should be able to destroy the old work once it's merged--just move over its license pools to the new work and recalculate the new work.

It might be worth it to also take another look at Work.merge_into and Work.similarity_to. This code contains very old code for determining if two Works are "the same", and I'm not really comfortable with it now that we use permanent work ID to make that decision instead.

### PR DESCRIPTION
**Pull request by aslagle**
Work.merge_into isn't used anymore, and Work.similarity_to is only used in one other obsolete piece of code that I will also remove.

Fixes #241

### CODE REVIEW

File: migration/20160516-remove-work-was-merged-into.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+DELETE FROM works where was_merged_into_id is not null;
+ALTER TABLE works DROP COLUMN was_merged_into_id CASCADE;
```
Comment (aslagle) [inline migration/20160516-remove-work-was-merged-into.sql]: This drops the materialized views - what's the right way to create them again?

Comment (leonardr) [inline migration/20160516-remove-work-was-merged-into.sql]: Recreate the new views by copying your new versions of materialized_view_works.sql and materialized_view_works_workgenres.sql to the bottom of the migration script.

**Comment by leonardr** [comment]
:+1: once the migration script recreates the materialized views.

File: migration/20160516-remove-work-was-merged-into.sql (hunk 1 above)
Comment (aslagle) [inline migration/20160516-remove-work-was-merged-into.sql]: Done

**Comment by leonardr** [comment]
Looks good!

**Pull request merged**